x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java

```diff
@@ -411,7 +406,12 @@ public enum Cap {
         /**
          * Support for semantic_text field mapping
          */
-        SEMANTIC_TEXT_TYPE(EsqlCorePlugin.SEMANTIC_TEXT_FEATURE_FLAG);
+        SEMANTIC_TEXT_TYPE(EsqlCorePlugin.SEMANTIC_TEXT_FEATURE_FLAG),
+
+        /**
+         * Support simplified syntax for named parameters for field and function names.
+         */
+        NAMED_PARAMETER_FOR_FIELD_AND_FUNCTION_NAMES_SIMPLIFIED_SYNTAX(Build.current().isSnapshot());
```

Comment:
Does this actually work?! (removing a previous capability and adding a new one? I thought this is a one way process)
I guess this works because we want this bug fix to reach the same versions as the previous PR (that added the initial capability).
But if the CI checks pass (including the Serverless one) then it's fine by me.

Reply:
> removing a previous capability and adding a new one?

Good question, first time this happens. But I think this should work, yeh: if we port this to 8.16, then it doesn't matter, as to future releases it'll look as if we never had the old cap. If we don't, 8.17+ will run the test(s) (actually just one yaml we have now), not find the old capability (still existing in 8.16) and not run the test(s).

Reply:
It took me a little while to be brave enough to try it! :-) I wish I didn't have a yaml test for the old syntax, or the test framework don't run bwc related tests on snapshot features.

With the old capability, I came across a failure with `yamlRestCompatTest`, it runs the old 8.16 yaml test on main, and because of the syntax change, it failed. I have to skip that yaml test in `build.gradle`, after that `yamlRestCompatTest` passed. And the second failure is MixedCluster test, this time it runs the main test on 8.17, it failed again. So I decided to make a new capability, that's the only approach I can think of so far.

Reply:
I'll run the serverless CI.